Analytical_Engine/Modify/SetOutlineElements1D.cs

```diff
@@ -79,7 +80,27 @@ public static IOpening<TEdge> SetOutlineElements1D<TEdge>(this IOpening<TEdge> o
         public static IRegion SetOutlineElements1D(this IRegion region, IEnumerable<IElement1D> outlineElements)
         {
             IRegion r = region.GetShallowClone(true) as IRegion;
-            r.Perimeter = BH.Engine.Geometry.Compute.IJoin(outlineElements.Cast<ICurve>().ToList())[0];
+
+            if (outlineElements.Count() == 1)
```

Comment:
I would remove this `if` statement - what if the `outlineElements` contains one curve that is not closed? Better let it pass through `IJoin` regardless the number of curves and then do the checks.

Reply:
thinking here was if the curve was a single circle for example. Then no real point wrapping it in a polycurve. still ok with changing to what you suggest, but any thoughts about that?

Reply:
Right, I was convinced that `IJoin` returns the original curve in case of single item input.

I take back my comment, but now when I am thinking about it I am not sure if it would not be better to refine that in `Join` (redo it to return `ICurve` and return single item inputs without wrapping into `PolyCurve`) instead of adding such check here?

And if we are meant to keep the code as is, it would still be good to check if the input curve is closed. To wrap it a bit, maybe:

```
IEnumerable<ICurve> joinedCurves = outlineElements;
            if (outlineElements.Count() != 1)
                joinedCurves = Engine.Geometry.Compute.IJoin(outlineElements.Cast<ICurve>().ToList());

            if (joinedCurves.Count() == 1)
            {
                if (!joinedCurves.First().IIsClosed())
                    Engine.Reflection.Compute.RecordWarning("The outline elements assigned to the region do not form a closed loop.");

                r.Perimeter = joinedCurves[0];
            }
            else
            {
                Engine.Reflection.Compute.RecordWarning("The outline elements assigned to the region are disjointed.");
                r.Perimeter = new PolyCurve { Curves = outlineElements.Cast<ICurve>().ToList() };
            }
```

Not sure if I did not float with thoughts too far though 🤣

Reply:
updated